src/renderer/component/footer/footer.js

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import CacheButton from './cache.js';
+import Throttle from './throttle.js';
+import RequestCount from './request-count.js';
+import ProxyStatus from './proxy-status.js';
+import UpdateStatus from './update-status';
+
+import constants from 'common/constants.js';
+import {
+  toggleCaching,
+  toggleThrottling as toggleThrottle,
+  setThrottleRate,
+  clearRequests
+} from 'common/actions/proxy.js';
+
+import { getUpdateStatus } from '../../reducers/app.js';
+import { getRequestData } from '../../reducers/requests.js';
+import { getProxyState } from '../../reducers/proxy.js';
+
+const Footer = props =>
+  <div className="footer">
+    <CacheButton
+      cachingEnabled={props.cachingEnabled}
+      toggleCaching={props.toggleCaching}
+    />
+    <Throttle
+      toggleThrottle={props.toggleThrottle}
+      setThrottleRate={props.setThrottleRate}
+      throttleEnabled={props.throttleEnabled}
+      throttleRate={props.throttleRate}
+    />
+    <RequestCount
+      requestData={props.requestData}
+      clearRequests={props.clearRequests}
+    />
+    <ProxyStatus
+      proxyStatus={props.proxyStatus}
+      proxyReason={props.proxyReason}
+    />
+    <UpdateStatus
+      status={props.updateStatus}
+      info={props.updateInfo}
+    />
+  </div>;
+
+Footer.propTypes = {
+  cachingEnabled: PropTypes.bool.isRequired,
+  toggleCaching: PropTypes.func.isRequired,
+  proxyStatus: PropTypes.oneOf([
```

Comment:
Should we define these enumerations as their own types?
Otherwise, everywhere where we're referring to `proxyStatus`, we need this whole block of possible values

Reply:
Probably. Any preference where?

Reply:
Probably somewhere in `./src/common`?